### ISSUE TITLE
IsPublished only works with lowercase cultures (8.15)

### ISSUE THREAD

**Issue by patrickdemooij9** [umbraco/Umbraco-CMS#10695]
### Which *exact* Umbraco version are you using? For example: 8.13.1 - don't just write v8

8.15

### Bug summary

When using .IsPublished(culture) on a multi-language website, it only works when the culture has been set to lower case.

I think this has something to do with the MsgPack change or just the 8.15 update in general, as we haven't run into this issue before. I can also reproduce this issue on the latest version of the v8/contrib branch (which is still 8.15-rc).

Model.IsPublished("en-US") will return false, while Model.IsPublished("en-us") will return true.
![image](https://user-images.githubusercontent.com/11466511/125947568-6c6e5e41-c902-4df9-9b77-3127e22dc02b.png)

### Steps to reproduce

- Start with the starter kit
- Change the homepage document type so that it is multi-language. You don't have to change any of the properties
- Update the home.cshtml to show the following:
```
Published with culture: @Model.IsPublished(Culture)
<br />
Published with en-US: @Model.IsPublished("en-US")
<br />
Published with lower case en-US: @Model.IsPublished("en-us")
```
- Go to the homepage. You should now see that all 3 are shown with true. It could be that they aren't already, but I think this has something to do with the cache.
- Publish the homepage
- The first two will return false, while the last one will return true. Even though all 3 should be returning true.

### Expected result / actual result

IsPublished() should return true even if the culture is not all lower-case

---
_This item has been added to our backlog [AB#13193](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/13193)_

**Comment by DanDiplo** [comment]
You can change Message Pack to use the "old" JSON format to test your hypothesis by adding / changing the following key in `web.config`

`<add key="Umbraco.Web.PublishedCache.NuCache.Serializer" value="JSON"/>`

That way you'd see if it was down to how it's being serialised...

**Comment by nul800sebastiaan** [comment]
Thanks @patrickdemooij9 and @DanDiplo - that workaround also helps for now.

I've been looking in all kinds of places to see if casing can be preserved in MsgPack but I've not been able to find more than this discussion - https://github.com/neuecc/MessagePack-CSharp/issues/220

I'm not sure how MsgPack seems to mangle this string and I am definitely not sure if this is the only string that now gets returned in the wrong casing.

Hopefully @nzdev can shed some light on this?

**Comment by nikolajlauridsen** [comment]
Fixed in #10959 😄
